woocommerce/woocommerce
# blueprint - add php package and cli commands

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds experimental blueprint package and cli commands behind a feature flag for testing.

We'll revisit this package after the testing period to refactor and simplify the codebase if needed.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch. Make sure to install `WooCommerce Beta Tester`
2. Skip the core profiler.
3. Navigate to `Tools -> WCA Test Helper -> Features` and enable `blueprint` feature.
4. SSH into the server.
5. Execute `wp wc blueprint`
6. You should see usage for `wp wc blueprint export` and `wp wc blueprint import`
7. Run `wp wc blueprint export ./test.json` to export a test file.
8. Run `wp wc blueprint import ./test.json` 
9. Confirm the commands work without any errors.





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add blueprint behind a feature flag for testing purposes.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
